src/katgpucbf/fsim/main.py

```diff
@@ -312,15 +316,23 @@ async def async_main() -> None:
     descriptor_senders = [
         DescriptorSender(sender.stream, sender.descriptor_heap, SPEAD_DESCRIPTOR_INTERVAL_S) for sender in senders
     ]
-    descriptor_tasks = [asyncio.create_task(sender.run()) for sender in descriptor_senders]
+    for descriptor_sender in descriptor_senders:
+        tg.create_task(descriptor_sender.run())
 
     if args.main_affinity >= 0:
         os.sched_setaffinity(0, [args.main_affinity])
     sync_time = time.time()
-    await asyncio.gather(*(sender.run(sync_time, args.run_once) for sender in senders))
-    for sender in descriptor_senders:
-        sender.halt()
-    await asyncio.gather(*descriptor_tasks)
+    async with asyncio.TaskGroup() as sender_tg:
+        for sender in senders:
+            sender_tg.create_task(sender.run(sync_time, args.run_once))
+    for descriptor_sender in descriptor_senders:
+        descriptor_sender.halt()
```

Comment:
Appreciate the rename as well (from `sender in descriptor_senders`).

Reply:
Thank mypy - it didn't like having `sender` be used as the loop variable for two loops with different types.